api/daemon.go

```diff
@@ -129,42 +129,48 @@ func checkForUpdate() (bool, string, error) {
 }
 
 // applyUpdate downloads and applies an update.
-func applyUpdate(version string) (err error) {
+func applyUpdate(version string) error {
 	manifest, err := fetchManifest(version)
 	if err != nil {
-		return
+		return err
 	}
 
 	// Get the executable directory.
 	binDir, err := osext.ExecutableFolder()
 	if err != nil {
-		return
+		return err
 	}
 
-	// Initialize the updater object.
-	up, err := update.New().VerifySignatureWithPEM([]byte(developerKey))
-	if err != nil {
-		// should never happen
-		return
-	}
+	// Configure the update.
+	opts := new(update.Options)
+	opts.SetPublicKeyPEM([]byte(developerKey))
 
 	// Perform updates as indicated by the manifest.
 	for _, file := range manifest[1:] {
+		// set update path
+		opts.TargetPath = filepath.Join(binDir, file)
+
 		// fetch the signature
-		var sig []byte
-		sig, err = getHTTP(version, file+".sig")
+		opts.Signature, err = getHTTP(version, file+".sig")
```

Comment:
`getHTTP` returns a byte slice (response) and an error if the GET failed. Here, the GET is fetching the signature of the file to be downloaded, and storing it in the `opts` struct. This struct is later passed to `update.Apply`, which performs the actual update (including verifying the signature).

There is such a test in daemon_test.go. However, there isn't a valid signature test, since that's more of a pain.